scripts/irods/test/test_izonereport.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import print_function
+import sys
+if sys.version_info >= (2, 7):
+    import unittest
+else:
+    import unittest2 as unittest
+import os
+import json
+from .. import lib
+from . import session
+
+class Test_Izonereport(session.make_sessions_mixin([], []), unittest.TestCase):
+
+    def setUp(self):
+        super(Test_Izonereport, self).setUp()
+        with session.make_session_for_existing_admin() as admin:
+             admin.assert_icommand(['iadmin', "mkresc", "testResc", 'unixfilesystem', lib.get_hostname() + ":" + "/tmp/irods/testResc"], 'STDOUT_SINGLELINE', 'unixfilesystem')
+             admin.assert_icommand(['iadmin', "mkresc", "anotherResc", 'unixfilesystem', lib.get_hostname() + ":" + "/tmp/irods/anotherResc"], 'STDOUT_SINGLELINE', 'unixfilesystem')
+
+    def tearDown(self):
+        super(Test_Izonereport, self).tearDown()
+        with session.make_session_for_existing_admin() as admin:
+             admin.assert_icommand(['iadmin', "rmresc", "testResc"])
+             admin.assert_icommand(['iadmin', "rmresc", "anotherResc"])
+
+    def test_resource_json_has_id(self):
+        with session.make_session_for_existing_admin() as admin:
+            _, out, _ = admin.assert_icommand(['izonereport'], 'STDOUT')
+            zone_report = json.loads(out)
+            self.assertGreaterEqual(len(out_j["zones"][0]["icat_server"]["resources"]), 1)
+            for resource in out_j["zones"][0]["icat_server"]["resources"]:
+                self.assertTrue('id' in resource)
```

Comment:
Would it be worth verifying that the IDs are correct? I suppose that is more testing the resource property getter/setter than `izonereport`. Just a thought.

Also, minor: Are we going to be pulling `test_izonereport_with_coordinating_resources__ticket_3303` and `test_izonereport_key_sanitization` into this new suite? I recall talking about that, but maybe I am mistaken.

Reply:
yes, we did.    let's get the izonereport tests into a single well-named spot.

Reply:
Will do.

In regards to testing the validity of the ids, the c++ test I had written earlier covers this. I recall someone mentioning that the reason I was running into trouble was because I needed to use a new connection to call `rcServerReport`, so I can fix that up and commit it.

Reply:
Right, like you say, the issue from the C++ tests was that you were re-using the same connection in the administration library calls (which is certainly the intuitive thing to do). Unfortunately, the resource manager is not updated in the connected agent which is why the new resources were not appearing. In the python suite, we are using iCommands which results in a new connection for each call, so it should work if a query or `ilsresc` with some parsing is used.

Hope that helps!